platform/security/rest/security-rest-clientapi/src/main/java/org/codice/ddf/cxf/client/ClientFactoryFactory.java

```diff
@@ -222,6 +223,32 @@
       String clientSecret,
       String oauthFlow);
 
+  /**
+   * Returns an initialized SecureCxfClientFactory
+   *
+   * @param endpointUrl
+   * @param interfaceClass
+   * @param sourceId
+   * @param discoveryUrl
+   * @param clientId
+   * @param clientSecret
+   * @param username
+   * @param password
+   * @param additionalParameters
+   * @param <T>
+   * @return
+   */
+  <T> SecureCxfClientFactory<T> getSecureCxfClientFactory(
+      String endpointUrl,
+      Class<T> interfaceClass,
+      String sourceId,
```

Comment:
are sourceId and discoveryUrl specific to DDF's use of this not necessarily generic use of a secure client?

Reply:
The source id is specific to DDF. It uses that to store the tokens locally so we don't make a call to get tokens on every request. The discovery URL is not DDf specific. We need that to get the correct URL to send a request to. The discovery URL should be where the metadata is hosted.